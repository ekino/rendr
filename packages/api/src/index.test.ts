import Express from "express";

import { createApi } from "./index";
import { RequestCtx, createPage, Page } from "@ekino/rendr-core";

describe("test createApi", () => {
  it("with no valid page found by the loader", async () => {
    function loader(ctx: RequestCtx) {}

    const api = createApi(loader);
    const req = jest.fn<Express.Request, any[]>();
    const res = jest.fn<Express.Response, any[]>();
    // @ts-ignore
    res.set = jest.fn().mockReturnThis();

    const next = jest.fn<Express.NextFunction, any[]>();

    // @ts-ignore
    const result = await api(req, res, next);

    // @ts-ignore
    expect(res.set).toBeCalledWith(
      "X-Rendr-Content-Type",
      "rendr/octet-stream"
    );
    expect(result).toBeUndefined();
  });

  it("with valid page found by the loader", async () => {
    async function loader(ctx: RequestCtx) {
      return await createPage({});
    }

    const api = createApi(loader);
    const req = jest.fn<Express.Request, any[]>();
    const res = jest.fn<Express.Response, any[]>();
    // @ts-ignore
    res.set = jest.fn().mockReturnThis();
    // @ts-ignore
    res.json = jest.fn().mockReturnThis();

    const next = jest.fn<Express.NextFunction, any[]>();

    // @ts-ignore
    const result = await api(req, res, next);

    // @ts-ignore
    expect(res.set).toBeCalledWith("X-Rendr-Content-Type", "rendr/document");
    // @ts-ignore
    expect(res.json).toBeCalledWith(createPage({}));
  });
});
