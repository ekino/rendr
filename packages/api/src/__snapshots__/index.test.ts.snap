// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test createApi it return the redirection page 1`] = `
RedirectPage {
  "location": "/redirected-page",
  "statusCode": 200,
}
`;

exports[`test createApi with no valid page found by the loader 1`] = `
ResponsePage {
  "body": "{\\"statusCode\\":404,\\"type\\":\\"document\\",\\"template\\":\\"default\\",\\"cache\\":{\\"ttl\\":0},\\"head\\":{\\"titleTemplate\\":\\"%s\\",\\"defaultTitle\\":\\"-\\",\\"title\\":\\"-\\",\\"links\\":[],\\"htmlAttributes\\":{},\\"meta\\":[]},\\"blocks\\":[],\\"settings\\":{},\\"id\\":\\"id\\",\\"path\\":\\"\\"}",
  "headers": Object {
    "Cache-Control": "private, max-age=0, no-cache",
    "Content-Type": "application/json",
    "X-Rendr-Content-Type": "rendr/document",
  },
  "statusCode": 200,
}
`;

exports[`test createApi with valid page found by the loader 1`] = `
ResponsePage {
  "body": "{\\"statusCode\\":200,\\"type\\":\\"document\\",\\"template\\":\\"default\\",\\"cache\\":{\\"ttl\\":0},\\"head\\":{\\"titleTemplate\\":\\"%s\\",\\"defaultTitle\\":\\"-\\",\\"title\\":\\"-\\",\\"links\\":[],\\"htmlAttributes\\":{},\\"meta\\":[]},\\"blocks\\":[],\\"settings\\":{},\\"id\\":\\"id\\",\\"path\\":\\"\\"}",
  "headers": Object {
    "Cache-Control": "private, max-age=0, no-cache",
    "Content-Type": "application/json",
    "X-Rendr-Content-Type": "rendr/document",
  },
  "statusCode": 200,
}
`;
